### DISCUSSION TITLE
memory issue on startup in some browsers

### DISCUSSION THREAD

**Discussion by mithrendal** [losso3000/coppenheimer#1] [Q&A] [answered]
@losso3000 there is  a memory issue when using on windows edge/chrome or also on mac in firefox...

macos safari and chrome is working fine though

reproduce steps:
1. start in private / inkognito browsing to get a fresh install without any cached files
2. press on the "install AROS m68k roms"

<img width="433" alt="image" src="https://github.com/losso3000/coppenheimer/assets/17108995/36749946-400d-4d04-9b55-18cf5b9682b7">

in your memdump_do method 

https://github.com/losso3000/coppenheimer/blob/48e06e26ecbb474d841f7eda99ec1ec6314b191e/js/vAmiga_ui-modified.js#L587

browser  complains 😬

RangeError: source array is too long
    memdump_do [https://coppenheimer.heckmeck.de/js/vAmiga_ui-modified.js](https://coppenheimer.heckmeck.de/js/vAmiga_ui-modified.js:587)[:](https://coppenheimer.heckmeck.de/js/vAmiga_ui-modified.js:587)[587](https://coppenheimer.heckmeck.de/js/vAmiga_ui-modified.js:587)
    memdump [https://coppenheimer.heckmeck.de/js/vAmiga_ui-modified.js](https://coppenheimer.heckmeck.de/js/vAmiga_ui-modified.js:550)[:](https://coppenheimer.heckmeck.de/js/vAmiga_ui-modified.js:550)[550](https://coppenheimer.heckmeck.de/js/vAmiga_ui-modified.js:550)
    do_animation_frame https://coppenheimer.heckmeck.de/js/vAmiga_ui-modified.js:769

strange that the browsers react so differently on this ... maybe it is sort of a race condition ? Maybe insert a console.log() statement and output the length of both `memimage_data` and `memdump_buffer` in order to know what is going on when using windows browsers or macos-firefox.

normally this is rather an issue than a discussion🤓, maybe you like also to enable the github issues feature too ? 

coppenheimer is mindblowing ... cool stuff !!! enjoyed it very much

**Comment by losso3000** [comment]
Thanks a lot for the bug report! I've recevied a similar report already and couldn't reproduce it yet, this helps. At first I thought I didn't anticipate huge browser windows and hard-coded a stupid limit somewhere, will investigate.

PS: Issues are activated now, that wasn't deliberate :)

**Comment by losso3000** [comment, marked as answer]
Moved to issue https://github.com/losso3000/coppenheimer/issues/2
